### PR TITLE
multi-region effort: dynamic security group

### PR DESCRIPTION
**Pull request by bernadinm**
This enables the ability for users to dynamically append or delete security group entries to the internal security group. This will be a breaking change for the users because terraform does not process that the rule already exist and there is nothing to do.

This GH issue is being tracked here: https://github.com/terraform-providers/terraform-provider-aws/issues/7058

When a user plans to upgrade to this version which has the change, they will need to remove the existing rule and have terraform add it back in. Another option is to mutate a terraform.tfstate file, but this method  below works and is less risk averse IMHO.

```
AWS_DEFAULT_REGION=us-east-1 aws ec2 revoke-security-group-ingress --group-id $(terraform state show module.dcos.module.dcos-infrastructure.module.dcos-security-groups.aws_security_group.internal | grep -Eo sg-.\+\$ | tail -1) --ip-permissions '[{"IpProtocol": "-1", "IpRanges": [{"CidrIp": "172.12.0.0/16"}]}]'
AWS_DEFAULT_REGION=us-east-1 aws ec2 revoke-security-group-egress --group-id $(terraform state show module.dcos.module.dcos-infrastructure.module.dcos-security-groups.aws_security_group.internal | grep -Eo sg-.\+\$ | tail -1) --ip-permissions '[{"IpProtocol": "-1", "IpRanges": [{"CidrIp": "0.0.0.0/0"}]}]'
terraform apply
```

This command allows a user to see what is the current status of the security group.

```
AWS_DEFAULT_REGION=us-east-1  aws ec2 describe-security-groups --group-id $(terraform state show module.dcos.module.dcos-infrastructure.module.dcos-security-groups.aws_security_group.internal | grep -Eo sg-.\+\$ | tail -1)
```

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -33,20 +33,26 @@ resource "aws_security_group" "internal" {
...
                                 "Cluster", var.cluster_name))}"
+}
 
-  ingress {
-    from_port   = 0
-    to_port     = 0
-    protocol    = "-1"
-    cidr_blocks = ["${var.subnet_range}"]
-  }
+resource "aws_security_group_rule" "internal_ingress_rule" {
```
Comment (fatz) [inline main.tf:38]: @bernadinm lets document here why this change is needed. Can you please explain why we have to change from the previous all in one resource?

as `var.subnet_range` is a array and the user needs to know his networks beforehand anyways why do we have to introduce this breaking change.

Is the whole resource getting recreated on a change in this variable?

**Review by fatz — COMMENTED** [review]
Lets first document about this change and why we have to introduce it

File: main.tf (hunk 1 above)
Comment (bernadinm) [reply main.tf:38]: I've described it in detail here: terraform-providers/terraform-provider-aws#7058

but the TL;DR is that when you rerun `terraform apply` terraform thinks that everything that is in the `aws_security_group` should be there and nothing else. Which breaks when you add new regions to that internal group. Terraform will notice it and will remove it during the next terraform apply, and they will keep going back and forth of adding, and removing, etc.
